api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -246,6 +254,13 @@ private void maybeFileZendeskReviewRequest(Workspace workspace) {
 
   @Override
   public ResponseEntity<Workspace> createWorkspace(Workspace workspace) throws BadRequestException {
+    return ResponseEntity.ok(
+        recordOperationTime(() -> createWorkspaceImpl(workspace), "createWorkspace"));
```

Comment:
I had hoped to be able to pull live latency distributions from the traces to use. with Alerting. @blrubenstein looked into whether we could alert on Traces. and said the systems were basically totally separate.

I do see [they have alerting](https://cloud.google.com/trace/docs/trace-alerting), but it appears to be more limited to _the number of Stackdriver Trace spans ingested per month, your quota usage, and your rate of span ingestion_

For my purposes here, I basically needed quantities to send to verify the service is working for distributions.

Reply:
Ignore StackDriver tracing for now, I'm asking whether you can implement recording a request's latency using an interceptor, similar to the approach used in the code I linked, instead of writing new code on every method.

Reply:
Drive-by comment here: for some reason I'd been thinking we wouldn't be able to use an interceptor for logging per-inbound-request latency, but it looks to me like it would totally work.

The fancy stuff I'd done in https://github.com/all-of-us/workbench/pull/2901 editing Swagger templates was important for instrumenting *outbound* API requests where RW acts as a client to Firecloud, but for inbound APIs an interceptor does seem to have most of the information we'd need.

Reply:
No. I regret linking the TraceInterceptor, I used it as an example because it's the most similar to what you'd need to do.

Interceptors are a standard spring concept where you can install some pre and post-processing hook on a set of controller endpoitns. We have several such interceptors defined here: https://github.com/all-of-us/workbench/tree/2599364f3f32bacb1878854dae47ec8c61d08970/api/src/main/java/org/pmiops/workbench/interceptors

Why would we not want to record a general purpose latency metric split out by method name?

Reply:
I'm happy to do that on a later ticket. The priority with this branch is integrating and. demonstrating distribution metrics, and elapsed time was an easy starting point.